### PR TITLE
fix: use poetry, not old setuptools

### PR DESCRIPTION
**Pull request by LewisCowlesMotive**
This aims to solve the setuptools problem of #858 #856 and #855 but by using a more modern tool

`poetry build` should, AFAIK build a universal wheel for you.

https://github.com/LewisCowlesMotive/vcrpy/releases/tag/v6.0.2 has notes on testing this fix.

I don't know how your CI works, so there may yet be more adjustments there.

### CODE REVIEW

**Comment by graingert** [comment]
I think we should stick to setuptools for now. I'd merge a PR that switched to pyproject.toml based setuptools

**Comment by pjonsson** [comment]
@graingert is that what #767 provides, or did you have something else in mind?

**Comment by Lewiscowles1986** [comment]
I'll close today. Not interested in explicit dependency on setuptools

File: pyproject.toml [hunk 1]
```diff
@@ -1,3 +1,66 @@
...
+[tool.poetry.dependencies]
+python = ">=3.8"
+pyyaml = "^6.0.1"
+wrapt = "^1.16.0"
+yarl = "^1.9.4"
+urllib3 = [
+    { version = "<2", markers = "python_version < '3.10'" },
+    { version = "<2", markers = "platform_python_implementation == 'PyPy'" },
+    { version = ">=2.2.2", markers = "python_version >= '3.10' and platform_python_implementation != 'PyPy'" }
+]
```
Comment (LewisCowlesMotive) [inline pyproject.toml:39]: after thought. This is incorrect. This package has no dependencies as it is, itself a dev dependency.

**Pull request closed without merging**